docs/configuration.rst

```diff
@@ -160,6 +170,10 @@ creation. Otherwise defaults are assumed and only the specified user or group na
 `providers.cfg`_ and `permissions.cfg`_ for further details about specific formatting and behaviour of each available
 field.
 
+A section for webhooks has also been added to the combined configuration file. This section defines a list of
+urls that should be called when either creating or deleting a user. The webhooks urls are responsible for any extra
+steps that should be taken on external services during the user creation/deletion.
+
```

Comment:
"defines a list of
urls that should be called when either creating or deleting a user"

I thought this would be precise enough, as it indicates the webhooks are getting called when the user is created or deleted.
I guess I could be more precise? Indicating the webhooks specified for user creation will get called at the end of the user's creation process, and same thing for deletion. I also assumed it was implicit that the user doesn't already exists when we create a user, or that the user exists when we try to delete it, but I could add a comment for that?

Reply:
Also, there is no documentation for the webhook api yet, as the webhook are not precised for now. I would need more explicit info on what webhook will be used and their payload.

Reply:
I could add a comment the callback part, explaining that a magpie url will be sent along the webhook requests, which the webhook must callback if a problem occurs during its execution. Also, the callback will get executed if the webhook requests are wrong from the start (such as using an unavailable webhook url.)

Reply:
> I guess I could be more precise?

Yes it's about precision. We discuss Francis and I about when the webhook should be called in the user creation process and it has its importance. If the webhook is called before the user is created, the webhook handler cannot query Magpie about that user like creating some permission for it. I should have said "when webhooks are called **_during creation and deletion process_**". I agree my when wasn't really clear :P. 

> Also, there is no documentation for the webhook api yet, as the webhook are not precised for now

What do you mean? It's Magpie that define the webhook API. If someone want to be notified by Magpie, it must know that Magpie will do a POST on the given url with a particular payload, which is right now {"user_name": user_name, "temp_url": "temp_url:80/todo"}.

I was a little bit optimistic when Francis and I discussed about synchronous or async webhooks in a way that we could allow any existing app to be registered in Magpie, but they all required special payload. I look for example at the slack webhook (given that we want to notify a special channel upon user creation). But Slack required a specific POST with its own payload. So, unless we add in the configuration the url, the method and the payload template, I doubt that we will be able to notify third party app. @fmigneault I would like to hear from you too on this one. Maybe, it could be done fairly easily and be really versatile? Looking at https://api.slack.com/messaging/webhooks, we would need to put in the configuration : 
```
webhook:
    create:
        slack:
            url: https://hooks.slack.com/services/T00000000/B00000000/XXXXXXXXXXXXXXXXXXXXXXXX
            method: POST
            payload:
                "text": "A new user, {user_name}, has just been created on Magpie. Calling {temp_url}, can signal an issue with it!"
```

> I could add a comment the callback part, explaining that a magpie url will be sent along the webhook requests

And yes this part is also important, what is the `temp_url` in the payload and how tu use it!

Reply:
See my other comment, the above extra fields goes along what I mentioned. 
I would replace the structure to 
```
webhooks:
  - name: slack  # this us just to identify
    action: create|remove
    method: GET|POST|PUT|DELETE 
    url: <location>
    payload: ...
```

I like the idea to allow custom JSON payload, but we must be extremely careful about "exposed" values. 
Only `{user_name}` and `{temp_url}` should be allowed templates. We must avoid leaking credentials, so the template replacement must explicitly allow only those values.

Reply:
I prefer your structure too!
And I agree that the templating must check for a finite list of variable. I never thought otherwise.

Reply:
Available template variables should also be documented in the 'Webhook API'.